dev/com.ibm.ws.kernel.service/src/com/ibm/wsspi/kernel/service/utils/SerializableSchedule.java

```diff
@@ -215,44 +296,86 @@ public boolean equals(Object o) {
 
         //Compare fields taking into account possible null values
         boolean startupBool = Boolean.compare(this.startup, c.startup) == 0;
-        boolean validDays = this.validDays == null ? c.validDays == null : this.validDays.equals(c.validDays);
+        boolean validDaysBool = this.validDays == null ? c.validDays == null : (this.validDays.containsAll(c.validDays) && c.validDays.containsAll(this.validDays));
```

Comment:
Should the data type be a Set rather than a List? Then Set.equals could be used for comparison, and validDays.contains would also be more efficient.
I suppose the disadvantage to this would be if the value gets printed, it will look fairly random rather than orderly.  TreeSet (which is a SortedSet) could help with that.